packages/blockchain/bch/src/lib/services/bch.sdk.virtualAccount.ts

```diff
@@ -1,45 +1,11 @@
-import { PendingTransaction } from '@tatumio/api-client'
-// @ts-ignore
-import * as coininfo from 'coininfo'
-import { Currency } from '@tatumio/api-client'
-import { bchWallet } from '../bch.sdk.wallet'
-// @ts-ignore
-import { Transaction, TransactionBuilder, ECPair, ECSignature } from '@tatumio/bitcoincashjs2-lib'
-import { BchSdkError } from '../bch.sdk.errors'
-import { SdkErrorCode } from '@tatumio/shared-abstract-sdk'
+import { ApiServices, TransferBchKeyPair, TransferBchKMS, TransferBchMnemonic } from '@tatumio/api-client'
+import { abstractBlockchainVirtualAccount } from '@tatumio/shared-blockchain-abstract'
+import { Blockchain } from '@tatumio/shared-core'
 
-/**
- * Sign Bitcoin Cash pending transaction from Tatum KMS
- * @param tx pending transaction from KMS
- * @param mnemonic mnemonic to generate private keys to sign transaction with.
- * @param testnet mainnet or testnet version
- * @returns transaction data to be broadcast to blockchain.
- */
-export const signKmsTransaction = async (
-  tx: PendingTransaction,
-  mnemonic: string,
-  testnet: boolean,
-): Promise<string> => {
-  if (tx.chain !== Currency.BCH || !tx.withdrawalResponses) {
-    throw new BchSdkError(SdkErrorCode.UNSUPPORTED_CHAIN)
+export const virtualAccountService = (args: { blockchain: Blockchain }) => {
+  return {
+    ...abstractBlockchainVirtualAccount(args),
+    send: async (body: TransferBchMnemonic | TransferBchKeyPair | TransferBchKMS) =>
+      ApiServices.offChain.blockchain.bchTransfer(body),
```

Comment:
this should be if (body.signatureId) -> apiServices else {local offchain tx implementation)